index.js

```diff
@@ -79,6 +80,26 @@ class Schematic {
     const schem = sponge.write(this)
     return gzip(nbt.writeUncompressed(schem))
   }
+
+  /**
+   * similar to js map, loop over all schem blocks
+   * @param {(block: Block, pos: Vec3, index: number) => {}} callback
+   */
+  async map (callback) {
+    let counter = 0
+    const { startX, startY, startZ } = this.start()
+    const { endX, endY, endZ } = this.end()
+    for (let y = startY; y <= endY; y++) {
+      for (let z = startZ; z <= endZ; z++) {
+        for (let x = startX; x <= endX; x++) {
+          const pos = new Vec3(x, y, z)
+          const block = this.getBlock(new Vec3(x, y, z))
+          counter++
```

Comment:
what is the use case for this counter ? (just asking)

Reply:
removed